### PR TITLE
feat: the BS now provides a central configuration which can be access…

### PR DESCRIPTION
**Pull request by jekutzsche**
…ed by the clients via EPS

Non-critical configurations (HD URLs and abbreviation at the moment) are deployed via a JSON located in the code.

More critical configurations, on the other hand, are provided via secrets in the deployment.

Refs iris-connect/iris-backlog#245

### CODE REVIEW

File: iris-backend-service/src/main/java/iris/backend_service/config/ServiceEndpointsConfig.java [hunk 1]
```diff
@@ -19,15 +20,13 @@
...
+	private final ConfigurationRpcService configurations;
 
 	@Bean(name = ENDPOINT)
 	public CompositeJsonServiceExporter jsonRpcServiceImplExporter() {
 
 		var compositeJsonServiceExporter = new CompositeJsonServiceExporter();
-		compositeJsonServiceExporter.setServices(new Object[] { health, locations, alerts });
-		compositeJsonServiceExporter
-				.setServiceInterfaces(new Class<?>[] { HealthRPC.class, LocationRPC.class, MessageRPC.class });
-		compositeJsonServiceExporter.setAllowMultipleInheritance(true);
```
Comment (lucky-lusa) [inline iris-backend-service/src/main/java/iris/backend_service/config/ServiceEndpointsConfig.java:30]: Are you sure this is not necessary anymore?

File: iris-backend-service/src/main/java/iris/backend_service/configurations/CentralConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	private void initialize() throws IOException {
+
+		configurationData = objectMapper.registerModule(new JavaTimeModule())
+				.readValue(configResource.getInputStream(), ConfigurationData.class);
+	}
+
+	Optional<HdProxyConfig> getHdProxyConfigFor(String client) {
+
+		return configurationData.hdProxyConfigs.stream()
+				.filter(it -> client.equals(it.hdName))
```
Comment (lucky-lusa) [inline iris-backend-service/src/main/java/iris/backend_service/configurations/CentralConfiguration.java:40]: Shouldnt we use hdEpsCn for clarity?

File: iris-backend-service/src/main/resources/application-dev_env.properties [hunk 3]
```diff
@@ -8,3 +8,6 @@ iris.messages.zammad.environment=empty
 iris.messages.github.personal-access-token=DISABLED
 iris.messages.github.repo-org=empty
 iris.messages.github.repo-name=empty
+
+iris.tokens.datSalt=${random.value}
```
Comment (lucky-lusa) [inline iris-backend-service/src/main/resources/application-dev_env.properties:12]: If we set it random, there is no possibility to test? As long as the used salt is not written to logs.

File: iris-backend-service/src/main/java/iris/backend_service/configurations/CentralConfiguration.java (hunk 2 above)
Comment (jekutzsche) [reply iris-backend-service/src/main/java/iris/backend_service/configurations/CentralConfiguration.java:40]: The configuration of the abbreviations and URLs has nothing to do with EPS, but with the communication via the proxy, right?

File: iris-backend-service/src/main/resources/application-dev_env.properties (hunk 3 above)
Comment (jekutzsche) [reply iris-backend-service/src/main/resources/application-dev_env.properties:12]: I thought so too, but you can always overwrite the properties from the outside (system property or environment variable).

Comment (jekutzsche) [reply iris-backend-service/src/main/resources/application-dev_env.properties:12]: I got a warning after the push that a secret is checked in here. Even though this is one for Dev, I removed it this way.

**Review by lucky-lusa — APPROVED** [review]

**Pull request merged**